django/src/rdwatch/views/model_run.py

```diff
@@ -158,96 +163,136 @@ def get_queryset():
-                    BoundingBoxGeoJSON('evaluations__geom'),
-                )
+    # Only bother calculating the entire bounding box of this model run
+    # list if the user has specified a region. We don't want to overload
+    # PostGIS by making it calculate a bounding box for every polygon in
+    # the database.
+    if filters.region is not None:
+        aggregate |= queryset.defer('json').aggregate(
+            # Use the region polygon for the bbox if it exists.
+            # Otherwise, fall back on the site polygon.
+            bbox=Coalesce(
+                BoundingBoxGeoJSON('evaluations__region__geom'),
+                BoundingBoxGeoJSON('evaluations__geom'),
             )
-
-        if aggregate['count'] > 0 and not aggregate['results']:
-            raise NotFound({'page': f"Invalid page '{page}'"})
-        aggregate['next'] = (
-            replace_query_param(request.build_absolute_uri(), 'page', page + 1)
-            if limit and page * limit < aggregate['count']
-            else None
-        )
-        aggregate['previous'] = (
-            replace_query_param(request.build_absolute_uri(), 'page', page - 1)
-            if limit and page > 1
-            else None
         )
 
-        serializer = HyperParametersListSerializer(aggregate)
-        return Response(data=serializer.data)
+    if aggregate['count'] > 0 and not aggregate['results']:
+        raise ValidationError({'page': f"Invalid page '{page}'"})
+
+    return 200, aggregate
+
+
+@router.get('/{id}', response={200: HyperParametersDetailSchema})
+def get_model_run(request: HttpRequest, id: int):
+    values = get_queryset().filter(id=id).values_list('json', flat=True)
+
+    model_run = values[0]
+
+    # TODO: use a resolver instead.
+    # Temporary until https://github.com/vitalik/django-ninja/issues/610 is fixed
+    if model_run['region']:
+        model_run['region'] = {
+            'name': RegionSchema.resolve_name(model_run['region']),
+            'id': model_run['region']['id'],
+        }
+
+    return 200, model_run
+
+
+@router.post(
+    '/{hyper_parameters_id}/site-model',
+    response={201: int},
+)
+def post_site_model(
+    request: HttpRequest,
+    hyper_parameters_id: int,
+    site_model: SiteModel,
+):
+    hyper_parameters = get_object_or_404(HyperParameters, pk=hyper_parameters_id)
+    site_evaluation = SiteEvaluation.bulk_create_from_site_model(
+        site_model, hyper_parameters
+    )
+    return 201, site_evaluation.id
+
+
+@router.post(
+    '/{hyper_parameters_id}/region-model',
+    response={201: list[int]},
+)
+def post_region_model(
+    request: HttpRequest,
+    hyper_parameters_id: int,
+    region_model: RegionModel,
+):
+    hyper_parameters = get_object_or_404(HyperParameters, pk=hyper_parameters_id)
+    site_evaluations = SiteEvaluation.bulk_create_from_from_region_model(
+        region_model, hyper_parameters
+    )
+    return 201, [eval.id for eval in site_evaluations]
```

Comment:
As part of this, I moved these two endpoints from `site_model.py` since they fall under the same model run router.